### PR TITLE
Added temperature sensors values in Powermeter

### PR DESCRIPTION
**Pull request by florinmihut**
…mperature2)

    - added in interface a variable containing the public OCMF key (publicKeyOcmf)
    - added in config the following options: meter_timezone, meter_dst, SC, UV, UD
    - added support for setting the following parameters: meter_dst and meter_timezone
    - added support for V2 interface

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: modules/RsIskraMeter/src/main.rs [hunk 1]
```diff
@@ -629,6 +629,8 @@ impl ReadyState {
             signed_meter_value: None,
             var_signed: None,
             voltage_v_signed: None,
+            temperature_1: Option::None,
```
Comment (SirVer) [inline modules/RsIskraMeter/src/main.rs:632]: no need to specficy the full path (i.e. `Option::None`), `None` is enough.

**Review by SirVer — APPROVED** [review]
only one nit for rust code.

File: modules/LemDCBM400600/main/lem_dcbm_400600_controller.hpp [hunk 2]
```diff
@@ -78,17 +88,22 @@ class LemDCBM400600Controller {
 
 private:
     const std::unique_ptr<HttpClientInterface> http_client;
-    std::string meter_id;
+    std::string meter_id = "";
+    std::string public_key = "";
+    std::string public_key_ocmf = "";
+    std::string version = "";
```
Comment (Pietfried) [inline modules/LemDCBM400600/main/lem_dcbm_400600_controller.hpp:94]: std::string initializes to empty string by default

File: modules/LemDCBM400600/main/lem_dcbm_400600_controller.hpp [hunk 3]
```diff
@@ -78,17 +88,22 @@ class LemDCBM400600Controller {
...
+    bool trasaction_is_ongoing = false;
     Conf config;
     std::unique_ptr<LemDCBMTimeSyncHelper> time_sync_helper;
 
+    std::vector<std::string> split(const std::string& str, char delimiter);
     void fetch_meter_id_from_device();
     void request_device_to_start_transaction(const types::powermeter::TransactionReq& value);
     void request_device_to_stop_transaction(const std::string& transaction_id);
     std::string fetch_ocmf_result(const std::string& transaction_id);
     void convert_livemeasure_to_powermeter(const std::string& livemeasure, types::powermeter::Powermeter& powermeter);
```
Comment (Pietfried) [inline modules/LemDCBM400600/main/lem_dcbm_400600_controller.hpp:105]: not part of the PR but could also be defined as a free function

File: types/powermeter.yaml [hunk 4]
```diff
@@ -299,3 +299,10 @@ types:
           This is intended for meters that only support signing a collection of meter values.
         type: object
         $ref: /units_signed#/SignedMeterValue
+      temperature1:
```
Comment (Pietfried) [inline types/powermeter.yaml:302]: This looks to LEM specific to me. Not sure which temperatures it actually measures, but would it make sense to you to add more generic properties like:
* ambient_temperature
* internal_temperature
* conductor_temperature
* ...

**Review by Pietfried — CHANGES_REQUESTED** [review]

File: types/powermeter.yaml (hunk 4 above)
Comment (florinmihut) [reply types/powermeter.yaml:302]: I agree but since this is not defined by all the power meters, this has to cover everyone. 
What if I have 2 cable temps? Or a high and low temp? 
Starting to give them "actual functional" names might pollute the interface with a lot of unused variables.
I spoke with Cornelius as well and he proposed to use temp 1 and 2 and their meaning is Powermeter dependent. Since anyway it is - some have it, some have it not, some measure the ambient some measure the internal some measure the high and the low. 
You get the picture. 
So I would be in the favor of keeping them temp 1 and 2 and their meaning is Powermeter actual implementation dependent.

Comment (hikinggrass) [reply types/powermeter.yaml:302]: If it's really that opaque we could also just add an array of temperature values here, what if there's a third sensor for example :D

Comment (Pietfried) [reply types/powermeter.yaml:302]: > I agree but since this is not defined by all the power meters, this has to cover everyone.
> Starting to give them "actual functional" names might pollute the interface with a lot of unused variables

For this we have optional properties. There are a lot of properties in the powermeter already defined this way.

> What if I have 2 cable temps? Or a high and low temp?

As @hikinggrass pointed out, what if we have 3? That is indeed specific to the powermeter and should therefore not be covered in a generic interface like powermeter like this. An interface is meant to be required by other modules. How should other modules know what temp1 or temp2 or temp3 are, especially if they mean different things for different implementations. Defining it this way doesn't seem useful to me.

If we add it like this, Id agree with @hikinggrass and add an array of temperature values, but I would rather add more meaningful properties as described above, maybe defining a type for a temperature and adding optional low / high values to it as properties.

Comment (corneliusclaussen) [reply types/powermeter.yaml:302]: I think an array is a good idea

File: modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp [hunk 5]
```diff
@@ -197,8 +199,18 @@ void LemDCBM400600Controller::convert_livemeasure_to_powermeter(const std::strin
...
+                                         types::temperature::Temperature{data.at("temperatureL"), "temperatureL"}});
+    } else {
+        for (auto& sensor : *powermeter.temperatures) {
+            if (sensor.identification == "temperatureL") {
+                sensor.temperature = data.at("temperatureL");
+            } else if (sensor.identification == "temperatureH") {
+                sensor.temperature = data.at("temperatureH");
+            }
+        }
+    }
```
Comment (Pietfried) [inline modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp:213]: the `powermeter` argument passed to this function is default initialized and the properties are then set in this function. Therefore this `else` block is never executed and can be dropped. Its also not required to check for `has_value()` in L202.

Apart from that I'd prefer a function signature like:
`types::powermeter::Powermeter LemDCBM400600Controller::convert_livemeasure_to_powermeter(const std::string& livemeasure)` that returns the powermeter object instead of passing it as an argument, but since this is legacy code im ok with keeping the signature like that

**Review by Pietfried — CHANGES_REQUESTED** [review]

File: modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp (hunk 5 above)
Comment (hikinggrass) [reply modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp:213]: But isn't powermeter.temperatures an optional vector so that check is still needed?

Comment (Pietfried) [reply modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp:213]: maybe I miss something but afaics the `temperatures` property is always not set so the else block is never executed

Comment (Pietfried) [reply modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp:213]: because the passed `powermeter` is default initialized and passed to this function

Comment (hikinggrass) [reply modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp:213]: ah you're right, then it's a good idea to drop the else block indeed!

Comment (florinmihut) [reply modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp:213]: You're right, I didn't check where the powermeter value comes from. 
Since it is created by the calling function the else branch would never execute.
Thank you for catching that.
I made the changes, including the signature of the function since I am changing it again and the change is minor.

**Review by Pietfried — APPROVED** [review]

**Review by hikinggrass — APPROVED** [review]
